modules/core/src/test/java/org/apache/ignite/internal/processors/cache/query/continuous/GridCacheContinuousQueryAbstractSelfTest.java

```diff
@@ -1205,6 +1215,171 @@ public void testExpired() throws Exception {
         }
     }
 
+    /**
+     * @throws Exception if failed.
+     */
+    @Test
+    public void testContinuousQueryRemoteFilterFactoryWithoutLocalLsnr() throws Exception {
+        doContinuousQueryRemoteFilterFactoryWithoutLocalLsnr(true);
+
+        doContinuousQueryRemoteFilterFactoryWithoutLocalLsnr(false);
+    }
+
+    /**
+     * @throws Exception if failed.
+     */
+    @Test
+    public void testContinuousQueryRemoteFilterWithoutLocalLsnr() throws Exception {
+        doContinuousQueryRemoteFilterWithoutLocalLsnr(true);
+
+        doContinuousQueryRemoteFilterWithoutLocalLsnr(false);
+    }
+
+    /**
+     * @throws Exception if failed.
+     */
+    @Test
+    public void testContinuousQueryRemoteTransformerWithoutLocalLsnr() throws Exception{
+        doContinuousQueryRemoteTransformerWithoutLocalLsnr(true);
+
+        doContinuousQueryRemoteTransformerWithoutLocalLsnr(false);
+    }
+
+    /**
+     * @throws Exception if failed.
+     */
+    @Test
+    public void testContinuousQueryInitialQueryWithoutLocalLsnr() throws Exception {
+        doContinuousQueryInitialQueryWithoutLocalLsnr(true);
+
+        doContinuousQueryInitialQueryWithoutLocalLsnr(false);
+    }
+
+    /**
+     * @throws Exception if failed.
+     * @param bypassFilter Whether remote filter should be bypassed.
+     */
+    private void doContinuousQueryRemoteFilterFactoryWithoutLocalLsnr(boolean bypassFilter) throws Exception {
+        filtered.set(false);
+
+        ContinuousQuery<Integer, Integer> qry = new ContinuousQuery<>();
+
+        qry.setRemoteFilterFactory(
+            FactoryBuilder.factoryOf((CacheEntryEventSerializableFilter<Integer, Integer>)evt -> {
+                filtered.set(true);
+
+                return bypassFilter;
+            }));
+
+        QueryCursor<Cache.Entry<Integer, Integer>> qryCursor = grid(0).cache(DEFAULT_CACHE_NAME).query(qry);
+
+        grid(0).<Integer, Integer>cache(DEFAULT_CACHE_NAME).put(1, 1);
+
+        assertTrue(GridTestUtils.waitForCondition(() -> filtered.get(), 100));
```

Comment:
Please, move the timeout to separate variable here and in another waitForCondition places.